src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4703,6 +4705,10 @@ def update_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         image_version.publishing_profile.replica_count = replica_count
     if image_version.storage_profile.source is not None:
         image_version.storage_profile.os_disk_image = image_version.storage_profile.data_disk_images = None
+    # target extended locations will be updated when --target-edge-zones is specified
+    if target_edge_zones is not None:
+        image_version.publishing_profile.target_extended_locations = target_edge_zones \
+            if len(target_edge_zones) > 0 else None
```

Comment:
It is recommended to tell users to enter `--target-edge-zones None` in the help message to clear the existing target edge zones